lib/packets/resultset_header.js

```diff
@@ -9,7 +9,7 @@ function ResultSetHeader (packet, bigNumberStrings)
   } else {
     this.fieldCount = packet.readInt8(); // skip OK byte
     this.affectedRows = packet.readLengthCodedNumber(bigNumberStrings);
-    this.insertId = packet.readLengthCodedNumber(bigNumberStrings);
```

Comment:
for values that fit into JS number ( like `-123` )we need to convert to js number, something like this:

```
    var insertIdString = insertId.toString(10);
    var insertIdNumber = insertId.toNumber();
    if (bigNumberStrings) {
      this.insertId = insertIdString;
    } else if (insertIdNumber.toString() === insertIdString) {
      this.insertId = insertIdNumber;
    } else {
      this.insertId = insertIdString;
    }
```